### PR TITLE
arrays-multidimensional_arrays-hard-q9

### PR DESCRIPTION
**Pull request by KaiKhademi**
## PR Checklist
- [x]  Are helper methods really needed here?
    > Opt for hardcoding input/output into the input provider when possible
- [x]  Are the assertion fail messages good?
    - [x]  Logical (does it make sense?)
    - [x]  Readable (can a student understand what’s wrong?)
    - [x]  Not too feed-y (does it give away the answer?)
    - [x]  Grammatically sound (is the grammar and punctuation in the sentence proper?)
    - [x]  Specific (does the fail message relate to the question?)
- [x]  Is the code formatted well?
    > Run `CTRL` + `ALT` + `L` in IntelliJ
- [x]  EOF newlines
- [x]  Do the comments for Parsons follow the correct convention?
- [x]  Is the question being tested sufficiently?
    - [x]  Are there representative cases to test the program?
    - [x]  What about edge cases? (empty strings, negatives, infinity, off-by-one errors, etc)
        > If the question doesn’t specify what to do, mark it as a bad question
- [x]  Are the test method names good?
    > Test method names show up in a “What went well” section on the website, so “mathTest” will become “What went well: math test”, which doesn’t make much sense. A better name in this example would be “calculatesTaxCorrectly” so it shows up as “What went well: calculates tax correctly”
---
## Website
- [x] Is the question text on the Gamification document up to date? 
    > Make sure to include the sample output if there is one for the question

### CODE REVIEW

File: src/arrays/multidimensional_arrays/hard/q9/ImAWholeDimensionAhead.java [hunk 1]
```diff
@@ -0,0 +1,28 @@
+package arrays.multidimensional_arrays.hard.q9;
+
+/*
+Create a java program called ImAwholeDimensionAhead. This program should have a main method and method called
```
Comment (SamamthaH) [inline src/arrays/multidimensional_arrays/hard/q9/ImAWholeDimensionAhead.java:4]: change to:
Make a java program that defines a 3D array in the main method and then passes it to a method called imAWholeDimensionAhead. This method should accept a 3D array and print out each element on the same line.

File: src/arrays/multidimensional_arrays/hard/q9/ImAWholeDimensionAhead.java [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+imAwholeDimensionAhead. This method should accept a 3D array as its parameter and print out each element to the System.
+The main method should call this method with any 3D array of your choice.
+
+ */
+
+public class ImAWholeDimensionAhead {
+    public static void main(String[] args) {
+        int[][][] nums =
+                {{{1, 2}, {3, 4}, {5, 6}},
```
Comment (SamamthaH) [inline src/arrays/multidimensional_arrays/hard/q9/ImAWholeDimensionAhead.java:13]: format this code

File: src/arrays/multidimensional_arrays/hard/q9/MainTest.java [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+    public Clause[] testSentence() {
+        String st = "";
+        for (int i = 0; i < nums.length; i++) {
+            for (int j = 0; j < nums[i].length; j++) {
+                for (int k = 0; k < nums[i][j].length; k++)
+                    st += nums[i][j][k] + " ";
+            }
+        }
+        return new Clause[]{
+                new StringLiteral(st.trim()),
```
Comment (SamamthaH) [inline src/arrays/multidimensional_arrays/hard/q9/MainTest.java:31]: get rid of the trim and optional space since this is a parsons and it will always have the extra space at the end

File: src/arrays/multidimensional_arrays/hard/q9/MainTest.java [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+                nums,
+                new int[][][]{{{1, 2}, {3, 4}}, {{5, 6}, {7, 8}}, {{9, 10}, {11, 12}}},
+                test
+        );
+    }
+
+    @ParameterizedTest
+    @MethodSource("inputProvider")
+    void correctImAWholeDimensionAheadMethod(int[][][] a) throws Throwable {
+        TestOption.incorrectStructureErrorMessage = "Your imAWholeDimensionAhead does not print the elements od 3D array correctly.";
```
Comment (SamamthaH) [inline src/arrays/multidimensional_arrays/hard/q9/MainTest.java:56]:
```suggestion
        TestOption.incorrectStructureErrorMessage = "Your imAWholeDimensionAhead does not print the elements of the 3D array correctly.";
```

File: src/arrays/multidimensional_arrays/hard/q9/MainTest.java [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+                new StringLiteral(ans)
+        };
+        Object[][] arguments = {
+                {a, int[][][].class}
+        };
+        MethodTest m = new MethodTest(ImAWholeDimensionAhead.class, "imAWholeDimensionAhead", arguments, methodSentence);
+        m.callMethod();
+
+    }
+}
```
Comment (SamamthaH) [inline src/arrays/multidimensional_arrays/hard/q9/MainTest.java:74]: add EOF line

**Review by SamamthaH — CHANGES_REQUESTED** [review]

File: src/arrays/multidimensional_arrays/hard/q9/MainTest.java [hunk 6]
```diff
@@ -0,0 +1,70 @@
...
+                nums,
+                new int[][][]{{{1, 2}, {3, 4}}, {{5, 6}, {7, 8}}, {{9, 10}, {11, 12}}},
+                test
+        );
+    }
+
+    @ParameterizedTest
+    @MethodSource("inputProvider")
+    void correctImAWholeDimensionAheadMethod(int[][][] a) throws Throwable {
+        TestOption.incorrectStructureErrorMessage = "Your imAWholeDimensionAhead does not print the elements of the 3D array correctly.";
```
Comment (SamamthaH) [inline src/arrays/multidimensional_arrays/hard/q9/MainTest.java:53]:
```suggestion
        TestOption.incorrectStructureErrorMessage = "Your imAWholeDimensionAhead method does not correctly print the elements of the 3D array.";
```

**Review by SamamthaH — APPROVED** [review]
